src/OpenApi/Factory/OpenApiFactory.php

```diff
@@ -279,38 +279,41 @@ private function collectPaths(ApiResource $resource, ResourceMetadataCollection
                 }
             }
 
+            $existingResponses = $openapiOperation?->getResponses() ?: [];
             // Create responses
             switch ($method) {
                 case HttpOperation::METHOD_GET:
                     $successStatus = (string) $operation->getStatus() ?: 200;
                     $responseContent = $this->buildContent($responseMimeTypes, $operationOutputSchemas);
-                    $openapiOperation = $openapiOperation->withResponse($successStatus, new Response(sprintf('%s %s', $resourceShortName, $operation instanceof CollectionOperationInterface ? 'collection' : 'resource'), $responseContent));
+                    $openapiOperation = $openapiOperation->withResponse($successStatus, $this->buildOpenApiResponse($existingResponses, $successStatus, sprintf('%s %s', $resourceShortName, $operation instanceof CollectionOperationInterface ? 'collection' : 'resource'), $responseContent));
```

Comment:
Actually the rule to apply here is that if a response exists in `$openapiOperation->getResponses()[200]` we should use it instead of building our own response. 
The code here should be:

```
if (!isset($existingResponses[$successStatus])) {
                    $openapiOperation = $openapiOperation->withResponse($successStatus, new Response(sprintf('%s %s', $resourceShortName, $operation instanceof CollectionOperationInterface ? 'collection' : 'resource'), $responseContent));
}
```

This allows the user to create his own responses and we don't alter the user definition, which may cause confusion. 

Obviously using a method for this can reduce the code complexity (especially bellow where we have mutliple statuses).

Reply:
Yeah it make sense.